### PR TITLE
Scheduler Improvements

### PR DESCRIPTION
**Pull request by deelawn**
Overview:
These changes are intended to make it easier to schedule new future network tasks; everything should be able to use either the `lastdata` or `syncrange` runners.  This requires there to be a `sub_kind` defined that allows us to run each of these runners for different kinds of tasks for the same network.  For example, beacon can have `lastdata` runners for rewards and for validator statuses.  There is also a concept of network parameters that makes it possible to inject any custom parameters into the request that will later be parsed by the worker.  A similar interface could be added or expanded upon if some other custom behavior needs to be defined later.

Highlights:
- Added the `sub_kind` to the UI.
- Removed validator status specific changes from my previous PR.
- Removed previous DB migrations and added new ones it their place (validator migrations were never run).
- Added `SubKind`.
- Added the network `ParamBuilder` interface and a beacon implementation.
- Added `FirstRequest` flag to task requests; this is set to true the first time a request is issued for a schedule.  It's a concept currently used by the eth worker but may be useful in the future.

### CODE REVIEW

File: cmd/scheduler-migration/migrations/000007_create_schedule_lastvalidatorstatuses_table.up.sql [hunk 1]
```diff
@@ -1,32 +0,0 @@
-CREATE TABLE IF NOT EXISTS schedule_lastvalidatorstatuses
```
Comment (pacmessica) [inline cmd/scheduler-migration/migrations/000007_create_schedule_lastvalidatorstatuses_table.up.sql:1]: Has this migration already been deployed to prod?

File: cmd/scheduler-migration/migrations/000007_add_sub_kind_to_schedule.up.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE schedule ADD COLUMN sub_kind TEXT;
+UPDATE schedule SET sub_kind = '';
```
Comment (pacmessica) [inline cmd/scheduler-migration/migrations/000007_add_sub_kind_to_schedule.up.sql:2]: so we should set `sub_kind` for existing tasks eh? haven't looked at manager or worker changes yet, do they expect this to be set?

File: networks/param_builder.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package networks
+
+import "github.com/figment-networks/indexer-scheduler/cmd/scheduler/config"
+
+// Params is a map of network specific parameters meant to be processeed by a worker.
+type Params map[string]interface{}
+
+// ParamBuilder stores all of the data necessary to build parameters for each network.
+type ParamBuilder struct {
```
Comment (pacmessica) [inline networks/param_builder.go:9]: nice!

File: persistence/postgresstore/postgresstore.go [hunk 4]
```diff
@@ -25,7 +25,13 @@ func NewDriver(db *sql.DB) *Driver {
 }
 
 func (d *Driver) GetConfigs(ctx context.Context) (rcs []structures.RunConfig, err error) {
-	rows, err := d.db.QueryContext(ctx, "SELECT id, run_id, network, chain_id, version, duration, kind, task_id, enabled, status, config FROM schedule")
+	rows, err := d.db.QueryContext(
+		ctx,
+		`
```
Comment (pacmessica) [inline persistence/postgresstore/postgresstore.go:30]: so all the spaces and newlines are included as part of the query, I've heard it's slightly more efficient to do it this way instead:

https://github.com/figment-networks/indexer-scheduler/blob/7ba5306fa0228ca6af73d2fbf7c9622f21edcc17/runner/lastrewards/persistence/postgresstore/postgresstore.go#L37-L39

File: cmd/scheduler-migration/migrations/000007_add_sub_kind_to_schedule.up.sql (hunk 2 above)
Comment (deelawn) [reply cmd/scheduler-migration/migrations/000007_add_sub_kind_to_schedule.up.sql]: Thanks for asking about that. I forgot to go back and fix that.  I modified this to be more correct where the column is defined as `not null` with a default value of `''`

**Pull request merged**